src/main/scala/io/radanalytics/streaming/amqp/AMQPBodyFunction.scala

```diff
@@ -26,7 +26,7 @@ import org.apache.qpid.proton.message.Message
   *
   * @tparam T
   */
-class AMQPBodyFunction[T] extends Function1[Message, Option[T]] with Serializable {
+class AMQPBodyFunction[T] extends ((Message) => Option[T]) with Serializable {
```

Comment:
This is better syntax, but generally good PR style to avoid making changes that aren't critical to the PR. Making non-critical changes increases chance of unnecessary merge conflicts.

Reply:
I agree, I just wanted to prepare the repo for releasing and remove the warnings during the build, this 1 was just suggested by idea... This syntax isn't backward compatible with old Scala, right?